src/bindings/python/tests/test_runtime/test_model.py

```diff
@@ -598,6 +599,29 @@ def check_shape(new_shape):
         in str(e.value)
     )
 
+def test_reshape_with_list_of_shapes():
+    # Model with three inputs
+    shape_a = [4, 4]
+    shape_b = [4, 4]
+    shape_c = [4, 4]
+    param_a = ops.parameter(Shape(shape_a), dtype=np.float32, name="A")
+    param_b = ops.parameter(Shape(shape_b), dtype=np.float32, name="B")
+    param_c = ops.parameter(Shape(shape_c), dtype=np.float32, name="C")
+    # Use a simple output node that combines all inputs
+    output = ops.add(param_a, param_b)
+    output = ops.add(output, param_c)
+    model = Model(output, [param_a, param_b, param_c])
+    # New shapes to assign
+    model.reshape({0: [2, 2], 1: [2, 2], 2: [2, 2]})
+    inputs = model.inputs    
+    new_shapes = [[2, 2], [2, 2], [2, 2]]
+    assert inputs[0].partial_shape == PartialShape(new_shapes[0])
+    assert inputs[1].partial_shape == PartialShape(new_shapes[1])
+    assert inputs[2].partial_shape == PartialShape(new_shapes[2])
```

Comment:
The test is good, but I think we need a few more cases, such as:

- reshaping only some shapes, ex. `model.reshape({0: [2, 2], 2: [2, 2]})`
- dynamic rank input shapes, ex. `PartialShape.dynamic()`
- dynamic dimensions input shapes, ex. `PartialShape(Dimension.dynamic(), 2)`
- interval input shapes, ex. `PartialShape((1, 3), (3, 5))`